front/lib/api/assistant/global_agents/configurations/dust/augment-dust-deep-2.ts

```diff
@@ -0,0 +1,103 @@
+import { getMainAgent } from "@app/lib/api/assistant/global_agents/configurations/dust/augment-dust-deep";
+import {
+  getDataSourceFileSystemAction,
+  getDataWarehousesAction,
+} from "@app/lib/api/assistant/global_agents/configurations/dust/dust-deep";
+import { getDataSourceViewIdsFromActions } from "@app/lib/api/assistant/permissions";
+import type { Authenticator } from "@app/lib/auth";
+import { isRemoteDatabase } from "@app/lib/data_sources";
+import { DataSourceViewResource } from "@app/lib/resources/data_source_view_resource";
+import { MCPServerViewResource } from "@app/lib/resources/mcp_server_view_resource";
+import type { AgentMessageType, Result, UserMessageType } from "@app/types";
+import { Err, Ok } from "@app/types";
+import type { AgentConfigurationType } from "@app/types/assistant/agent";
+
+export async function replaceCompanyDataActions(
+  auth: Authenticator,
+  agentConfiguration: AgentConfigurationType,
+  mainAgent: AgentConfigurationType
+) {
+  const dsViews = await DataSourceViewResource.fetchByIds(
+    auth,
+    getDataSourceViewIdsFromActions(mainAgent.actions)
+  );
+  const spaceIds = [...new Set(dsViews.map((item) => item.space))];
+  const allSpacesViews = await DataSourceViewResource.listBySpaces(
+    auth,
+    spaceIds
+  );
+
+  // Replace the datasource and datawarehouses actions with the new ones
+  agentConfiguration.actions = agentConfiguration.actions.filter(
+    (action) =>
+      action.sId !== "dust-deep-data-source-file-system-action" &&
+      action.sId !== "dust-deep-data-warehouses-action"
+  );
+
+  const dataSourceFileSystemMCPServerView =
+    await MCPServerViewResource.getMCPServerViewForAutoInternalTool(
+      auth,
+      "data_sources_file_system"
+    );
+
+  const newDataSourceFileSystemAction = getDataSourceFileSystemAction(
+    allSpacesViews
+      .filter((item) => item.dataSource.connectorProvider !== "webcrawler")
+      .map((item) => item.toJSON()),
+    auth.getNonNullableWorkspace().sId,
+    dataSourceFileSystemMCPServerView
+  );
+
+  const dataWarehousesMCPServerView =
+    await MCPServerViewResource.getMCPServerViewForAutoInternalTool(
+      auth,
+      "data_warehouses"
+    );
+
+  const newDataWarehousesAction = getDataWarehousesAction(
+    allSpacesViews
+      .filter((item) => isRemoteDatabase(item.dataSource))
+      .map((item) => item.toJSON()),
+    auth.getNonNullableWorkspace().sId,
+    dataWarehousesMCPServerView
+  );
+
+  if (newDataSourceFileSystemAction) {
+    agentConfiguration.actions.push(newDataSourceFileSystemAction);
+  }
+
+  if (newDataWarehousesAction) {
+    agentConfiguration.actions.push(newDataWarehousesAction);
+  }
+}
+
+export async function augmentDustDeep(
+  auth: Authenticator,
+  agentConfiguration: AgentConfigurationType,
+  agentMessage: AgentMessageType,
+  userMessage: UserMessageType
+): Promise<Result<AgentConfigurationType, Error>> {
```

Comment:
As discussed IRL, this version completely replaces dsfs/dw actions with equivalent actions scoped on the entire space of the original agent. This removes access to global space - deep only has access to the restricted space

Reply:
Wait we _remove_ the global space ?

Reply:
@PopDaph opened an issue to discuss about this (I can't find it right now) - but yes the idea was that we don't change the scope the agent. An agent that was on space X only would have access to the whole space X, but not other spaces. Accessing the company space would not be consistent with how the main agent was scoped

Reply:
pretty bearish on that. Makes everything super complex for 0 good reason

Reply:
Issue is not opened there's just a doc on Notion where we started discussing the options - 
We want to do what will be the most logic/expected behavior - I think it's the one that is the most easy to explain.

Reply:
Let's discuss it at 2pm cause there could be use cases for both options - taking into account that now agents can be into multiple spaces.